hadoop-hdds/common/src/main/java/org/apache/hadoop/util/MetricUtil.java

```diff
@@ -60,7 +75,15 @@ public static <T, E extends IOException> T captureLatencyNs(
     try {
       return block.get();
     } finally {
-      latencySetter.accept(Time.monotonicNowNanos() - start);
+      executeMetricsUpdateAction(() ->
+              latencySetter.accept(Time.monotonicNowNanos() - start));
     }
   }
+
+  public static void executeMetricsUpdateAction(Runnable action) {
+    if (!executor.isTerminated() && !executor.isShutdown()) {
+      executor.submit(action);
+    }
```

Comment:
I'm not sure this is safe, executor can be shutdown between check and `submit`.  I think it would be better to set `CallerRunsPolicy` / `DiscardPolicy` to avoid exceptions.

Reply:
fixed